crates/astria-composer/src/executor/bundle_factory/mod.rs

```diff
@@ -153,14 +184,34 @@ impl BundleFactory {
                 })
             }
             Err(SizedBundleError::NotEnoughSpace(seq_action)) => {
-                // if the bundle is full, flush it and start a new one
-                self.finished.push_back(self.curr_bundle.flush());
-                self.curr_bundle
-                    .push(seq_action)
-                    .expect("seq_action should not be larger than max bundle size, this is a bug");
-                Ok(())
+                if self.finished.len() >= self.finished_queue_capacity {
+                    Err(BundleFactoryError::FinishedQueueFull {
+                        curr_bundle_size: self.curr_bundle.curr_size,
+                        seq_action,
+                    })
+                } else {
+                    self.finished.push_back(self.curr_bundle.flush());
+                    // if the finished queue is full after the flush mark it as such
+                    if self.finished.len() == self.finished_queue_capacity {
+                        self.full = true;
+                    }
+
+                    let sized_seq_action = self.curr_bundle.push_check(seq_action).expect(
+                        "seq_action should not be larger than max bundle size, this is a bug",
+                    );
```

Comment:
why are we doing this again? i think we enter this code block after succeeding this check right?

Reply:
The first `push_check` is on the original `curr_bundle`. If the seq action doesn't fit in it then curr is flushed into the `finished` queue and replaced with a fresh `SizedBundle`. The seq action is then pushed into the new `curr_bundle`. The push into a `SizedBundle` the checks in `push_check` have to be done first. This isn't different from what we had before - those checks happened at the start of the push call anyways.